### PR TITLE
Update “Development” infrastructure to run migrations upon deployment

### LINKED ISSUE #1595: Update “Development” infrastructure to run migrations upon deployment
### Issue description

Currently, when tickets are merged with migrations, not everyone is remembering to run migrations on development (which is updated with the new code). This is causing an issue where Development gives a 500 error. To resolve this, migrations need to be run. 

To safeguard against the oversight, we can update our “development” infrastructure so that upon a deploy, it also runs migrations. It will bypass the need to manually run migrations.  

### Acceptance criteria

- [ ] Update “development” infrastructure to run migrations upon a deploy

### Additional context

_No response_

### Links to other issues

_No response_

### PR DESCRIPTION
**Pull request by CocoByte**
## Ticket 1595

Resolves #1595

## Changes

- Added migration step to dev deployment workflow (deploy-development.yaml)

## Context for reviewers

Currently we cannot manually fire the action for deployment to development, so a trigger has been setup to run this action on a push to this branch.  The logs from this will be used to verify whether this code can be merged.

## Code Review Verification Steps

Check the logs for action "Build and deploy development for release" to ensure migrations run automatically once code is deployed

#### Satisfied acceptance criteria and met development standards

- [X] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [X] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [X] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: .github/workflows/deploy-development.yaml [hunk 1]
```diff
@@ -28,6 +49,9 @@ jobs:
       - name: Collect static assets 
         working-directory: ./src
         run: docker compose run app python manage.py collectstatic --no-input
+      - name: Run Django migrations
```
Comment (abroddrick) [inline .github/workflows/deploy-development.yaml:52]: Fyi this is the wrong spot. You want to do this after the push below (name: Deploy to cloud.gov sandbox)

Comment (CocoByte) [reply .github/workflows/deploy-development.yaml]: Updated

File: .github/workflows/deploy-development.yaml [hunk 2]
```diff
@@ -4,6 +4,13 @@
 name: Build and deploy development for release
 
 on:
+  workflow_dispatch:
```
Comment (abroddrick) [inline .github/workflows/deploy-development.yaml:7]: don't forget to delete this

File: .github/workflows/deploy-development.yaml [hunk 3]
```diff
@@ -12,6 +19,7 @@ on:
       
     branches:
       - main
+      - nl/1595-auto-run-migrations
```
Comment (abroddrick) [inline .github/workflows/deploy-development.yaml:22]: remove this too

**Review by abroddrick — APPROVED** [review]
Approving since we tested and this works, but make sure to remove the workflow dispatch, and pull_request sections as well as the push branch value that is set to your branch. I marked them with comments. Won't block for that.

**Review by abroddrick — APPROVED** [review]
Let the merging commence!

**Pull request merged**